fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/KeyWithValueExpression.java

```diff
@@ -178,7 +178,7 @@ private KeyExpression getInnerKey() {
     @Nonnull
     @Override
     public KeyExpression getChild() {
-        return getInnerKey();
+        return getKeyExpression();
```

Comment:
It looks like `getPlannerExpressionChildren` still returns the full key. Is that a problem?

Reply:
Immediately, it isn't. Technically, it's inconsistent but it doesn't end up mattering because `getPlannerExpressionChildren()` is only ever used by the experimental planner and my next PR for that removes the `PlannerExpression` interface (and therefore the `getPlannerExpressionChildren()` method) from `KeyExpression`.

I looked into fixing it here but it's actually kind of gross because of what's a reference and what isn't. I'm personally inclined to not worry about it and take this fix as-is.